utility/log.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+"""
+Implements CephCI logging interface.
+
+In this module, we implement a singleton LOG object that can be used by all components
+of CephCI. It supports the below logging methods
+
+    - error
+    - warning
+    - info
+    - debug
+
+Along with the above, it provides support for pushing events to
+
+    - local file
+    - logstash server
+"""
+import logging
+from copy import deepcopy
+from typing import Any, Dict
+
+from .config import TestMetaData
+
+LOG_FORMAT = "%(asctime)s - %(levelname)s - %(filename)s:%(lineno)d - %(message)s"
+
+
+class LoggerInitializationException:
+    pass
+
+
+class Log:
+    """CephCI Logger object to help streamline logging."""
+
+    def __init__(self, filename="") -> None:
+        """
+        Initializes the logging mechanism based on the inputs provided."""
+        self._logger = logging.getLogger(filename)
+        self._log_level = logging.DEBUG
+        self._log_dir = None
+        self.log_format = LOG_FORMAT
+        self.module_name = None
```

Comment:
You may want to add a method to check if logstash is enabled for writing the metadata...

Reply:
This we are checking as part of TestMetadata creation in run.py